crates/ruff_cli/src/lib.rs

```diff
@@ -280,17 +280,27 @@ fn check(args: CheckArgs, log_level: LogLevel) -> Result<ExitStatus> {
         }
 
         if !cli.exit_zero {
-            if cli.diff || fix_only {
+            if cli.diff {
                 if !diagnostics.fixed.is_empty() {
                     return Ok(ExitStatus::Failure);
                 }
-            } else if cli.exit_non_zero_on_fix {
-                if !diagnostics.fixed.is_empty() || !diagnostics.messages.is_empty() {
-                    return Ok(ExitStatus::Failure);
+            } else if fix_only {
+                if cli.exit_non_zero_on_fix {
+                    if !diagnostics.fixed.is_empty() {
+                        return Ok(ExitStatus::Failure);
+                    }
+                } else {
+                    return Ok(ExitStatus::Success);
                 }
             } else {
```

Comment:
I'm a bit confused. Isn't `diff` `--fix`  and `fix_only` `--fix-only`? If so, shouldn't the `cli.diff` and `fix_only` branches be identical?

Reply:
No, those aren't the same. `--fix`, `--fix-only`, and `--diff` are all different options.